### PR TITLE
Documentation improvements for ViewModel lifecycle and Android Presenter

### PR DESCRIPTION
**Pull request by nmilcoff**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)
Documentation update and minor code improvement.

### :arrow_heading_down: What is the current behavior?
Documentation for Android Presenter is very limited, also new lifecycle is not explicitly detailed anywhere.

### :new: What is the new behavior (if this is a feature change)?
Improved documentation.

### :boom: Does this PR introduce a breaking change?
No.

### :bug: Recommendations for testing
Read the docs.

### :thinking: Checklist before submitting

- [X] All projects build
- [X] Follows style guide lines ([code style guide](https://github.com/MvvmCross/MvvmCross#code-style-guidelines))
- [X] Relevant documentation was updated ([docs style guide](https://www.mvvmcross.com/documentation/contribute/mvvmcross-docs-style-guide))
- [X] Nuspec files were updated (when applicable)
- [X] Rebased onto current develop

### CODE REVIEW

File: docs/_documentation/fundamentals/viewmodel-lifecycle.md [hunk 1]
```diff
@@ -28,11 +29,11 @@ void ViewDisappeared();
 void ViewDestroy();
 ```
 
-The MvxViewController, MvxFragment(s), MvxActivity and the UWP views will call those methods open the platform specific events that are fired. This will give us a more refined control of the ViewModel and the state of its lifecycle. There may be binding that you want to update or resources to clean up, these lifecycle events can help with that.
+The MvxViewController, MvxFragment(s), MvxActivity and the UWP views will call those methods when the platform specific events are fired. This will give you a more refined control of the ViewModel and its state. There may be certain bindings that you want to update or resources that you want to clean up in these calls.
 
-It should be noted however that it is not 100% reliable but it should work for most of the apps. We don't know what you do in the lifecycle of your app and what could interfere with the called order of the viewmodel lifecycle events.
```
Comment (MarcBruins) [inline docs/_documentation/fundamentals/viewmodel-lifecycle.md]: Typo: What whether that

File: docs/_documentation/presenters/android-view-presenter.md [hunk 2]
```diff
@@ -29,27 +30,57 @@ The presenter uses a set of `PresentationAttributes` to define how a view will b
...
 
-Used as root of the App and root for other layouts like Fragments.
+Use this attribute if you want to display an Activity in your application. An Activity will be the root of your application and also can act as a host for fragments. Alongside the attribute, your view can customize the presentation by using these attribute properties:
+
+- Extras: Use this `Bundle` to add any extra parameters to the Activity Intent.
+- SharedElements: Consists on a `IDictionary<string, View>` that you can use to add shared view elements to the transition. When using the AppCompat version, the string keys are not relevant.
+
+### MvxFragmentPresentationAttribute
+
+A Fragment is hosted inside an Activity (or a fragment). By using this ViewPresenter, you can decide whether to make all of your screens Activities, or to use an Activity host and many Fragments inside of it. The framework will then help you setting up the navigation and backstack.
```
Comment (MarcBruins) [inline docs/_documentation/presenters/android-view-presenter.md]: This sentence seems weird to me: "will then help you setting up the navigation and backstack"

File: docs/_documentation/presenters/android-view-presenter.md [hunk 3]
```diff
@@ -29,27 +30,57 @@ The presenter uses a set of `PresentationAttributes` to define how a view will b
...
+
+The ViewPresenter supports also nested fragments in one level: This means you can show fragments inside of a Fragment without extending any code!.
```
Comment (MarcBruins) [inline docs/_documentation/presenters/android-view-presenter.md]: "!."

File: docs/_documentation/presenters/android-view-presenter.md [hunk 4]
```diff
@@ -29,27 +30,57 @@ The presenter uses a set of `PresentationAttributes` to define how a view will b
...
 
-### MvxTabLayoutPresentationAttribute
+- Title: Title for the ViewPager. It will also be used as Title for the TabLayout when using MvxTabLayoutPresentationAttribute.
+- ViewPagerResourceId: The resource id for the ViewPager that will be used as host.
 
-Extends `MvxFragmentPresentationAttribute` and puts the Fragment into a ViewPager with TabLayout.
+### MvxTabLayoutPresentationAttribute (AppCompat only)
 
-### MvxViewPagerFragmentPresentationAttribute
+This attribute extends `MvxViewPagerFragmentPresentationAttribute`, which means you can use all the properties it provedes to customize the presentation. use this attribute over a Fragment view class to display a fragment inside of a ViewPager with TabLayout and take advantage of even more customization with this property:
```
Comment (MarcBruins) [inline docs/_documentation/presenters/android-view-presenter.md]: "provedes"

File: docs/_documentation/presenters/android-view-presenter.md [hunk 5]
```diff
@@ -75,16 +106,20 @@ The presenter is completely extensible! You can override any attribute and custo
 You can also define new attributes to satisfy your needs. The steps to do so are:
 
 1. Add a new attribute that extends `MvxBasePresentationAttribute`
-2. Subclass `MvxAppCompatViewPresenter` and make it the presenter of your application in Setup.cs (by overriding the method `CreatePresenter`).
+2. Subclass `MvxAndroidViewPresente` or `MvxAppCompatViewPresenter` and make it the presenter of your application in Setup.cs (by overriding the method `CreatePresenter`).
```
Comment (MarcBruins) [inline docs/_documentation/presenters/android-view-presenter.md]: "MvxAndroidViewPresente"

**Comment by nmilcoff** [comment]
@MarcBruins thanks for flagging 👍! I'll make a new PR today fixing this

**Pull request merged**